homeassistant/components/modem_callerid/__init__.py

```diff
@@ -1 +1,72 @@
-"""The modem_callerid component."""
+"""The Modem Caller ID integration."""
+import asyncio
+import logging
+
+from phone_modem import PhoneModem, exceptions
+import serial
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_DEVICE
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DATA_KEY_API, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [SENSOR_DOMAIN]
+
+
+async def async_setup(hass: HomeAssistant, config):
+    """Set up the Modem Caller ID component."""
+    hass.data[DOMAIN] = {}
+
+    hass.async_create_task(
```

Comment:
Some other integrations guard against multiple importing via this check:
```
if DOMAIN in config and not hass.config_entries.async_entries(DOMAIN):
```

Reply:
Also how does this actually do anything? I see from the docs that this integration is configured via a sensor platform, so I don't think async_setup would see a DOMAIN value here?

Reply:
`async setup` is importing any yaml config that the user already has. Without this, the import never happens. I can use `if DOMAIN in config and not hass.config_entries.async_entries(DOMAIN):` when it's determined that we can't actually use more than one device.